module-system/sov-modules-api/src/default_signature.rs

```diff
@@ -123,6 +124,12 @@ pub struct DefaultPublicKey {
     pub(crate) pub_key: DalekPublicKey,
 }
 
+impl Hash for DefaultPublicKey {
```

Comment:
To double check: why do we opt to implement hash manually with the serialized bytes instead of relying on the derive macro that will implement it outside the scope of serialization?

For further context: I believe the main advantage of bringing `Hasher` to the table was to separate the two domains of serialization & in-memory hash table. We are having problems because we intersect the two, when we have no formal guarantee that a "keyable" type will have uniform serialization implementation whereas they might have the same memory representation.

It might be that we have strict requirements in our ZK tree that will require these items to use the storage implementation. Is that the reason? If so, I think its best to add an explicit comment describing that so future developers will immediately understand the decision

Reply:
We can't derive `Hash` for this type because `DalekPublicKey` doesn't implement `Hash`, so we have to write our own. In this case, it just so happens that the type implements `Hash` re-using some of its custom serialization logic under the hood. I don't see an issue with this, but maybe I'm just not quite getting what you're saying.

> we have no formal guarantee that a "keyable" type will have uniform serialization implementation whereas they might have the same memory representation.

I believe we do, don't we? `Eq` and `Hash` guarantee consistency with each other. In this case, `DalekPublicKey` implements `Eq` as well as `AsRef<[u8]>` and we can assume that `pubkey1 == pubkey2` iff pubkey1.as_bytes() == pubkey2.as_bytes()`.